dev/scripts/exglobal_archive_vrfy.py

```diff
@@ -14,38 +15,17 @@ def main():
 
     config = cast_strdict_as_dtypedict(os.environ)
 
-    # Instantiate the Archive object
+    # Instantiate the Archive task object
     archive = Archive(config)
 
-    # update these keys to be 3 digits if they are part of archive.task_config.keys
-    for key in ['OCNRES', 'ICERES']:
-        try:
-            archive.task_config[key] = f"{archive.task_config[key]:03d}"
-        except KeyError as ee:
-            logger.info(f"key ({key}) not found in archive.task_config!")
-
-    # Pull out all the configuration keys needed to run the rest of archive steps
-    keys = ['current_cycle', 'RUN', 'PSLOT', 'ROTDIR', 'PARMgfs',
-            'ARCDIR', 'MODE', 'DO_JEDIATMENS', 'DO_FIT2OBS', 'DO_JEDIATMVAR', 'FHMIN_GFS',
-            'DO_JEDISNOWDA', 'DO_AERO_ANL', 'DO_PREP_OBS_AERO', 'NET', 'MODE', 'FHOUT_GFS',
-            'FHMAX_HF_GFS', 'FHOUT_GFS', 'FHMAX_FITS', 'FHMAX', 'FHOUT', 'FHMAX_GFS', 'DO_GSISOILDA', 'DO_LAND_IAU']
-
-    archive_dict = AttrDict()
-    for key in keys:
-        try:
-            archive_dict[key] = archive.task_config[key]
-        except KeyError as ee:
-            logger.warning(f"WARNING: key ({key}) not found in archive.task_config!")
-
-    # Also import all COMIN* and COMOUT* directory and template variables
-    for key in archive.task_config.keys():
-        if key.startswith(("COM_", "COMIN_", "COMOUT_")):
-            archive_dict[key] = archive.task_config.get(key)
+    # Collect all archive variables in complete arch_dict for YAML templates
+    # Use static utility methods from ArchiveVrfyVars
+    arch_dict = ArchiveVrfyVars.get_all_yaml_vars(archive.task_config)
 
-    with chdir(config.ROTDIR):
+    # Pass arch_dict to configure_vrfy which will render the Jinja2 YAML
+    arcdir_set = archive.configure_vrfy(AttrDict(arch_dict))
```

Comment:
```suggestion
    arcdir_set = archive.configure_vrfy(arch_dict)
```
Let `ArchiveVrfyVars.get_all_yaml_vars` return as `AttrDict`